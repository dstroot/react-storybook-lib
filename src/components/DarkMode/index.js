import React from 'react';
import './styles.scss';

/**
 * DarkMode!
 * https://timellenberger.com/blog/operating-system-dark-mode-in-your-css?utm_source=share&utm_medium=ios_app
 * https://timellenberger.com/blog/dynamic-theming-with-styled-components-and-nextjs
 */

class DarkMode extends React.Component {
  state = {
    supportsColorScheme: false,
    isDarkMode: false,
    isLightMode: false,
  };

  componentDidMount() {
    const supportsColorScheme = window.matchMedia('(prefers-color-scheme)')
      .matches;
    const isDarkMode = window.matchMedia('(prefers-color-scheme: dark)')
      .matches;
    const isLightMode = window.matchMedia('(prefers-color-scheme: light)')
      .matches;
    this.setState({ supportsColorScheme, isDarkMode, isLightMode });
  }

  render() {
    const { supportsColorScheme, isDarkMode, isLightMode } = this.state;
    if (supportsColorScheme) {
      if (isDarkMode)
        return (
          <div className="bg-dark p-5">
            <p className="text-white">Your system is in dark mode!</p>
          </div>
        );
      if (isLightMode)
        return (
          <div className="bg-light p-5">
            <p className="text-black">Your system is in dark mode!</p>
          </div>
        );

      // The browser supports light/dark mode but can't infer anything from the system
      return <>Your system light/dark mode preference is unset!</>;
    }

    // The browser doesn't support light/dark mode
    return <>Your browser doesn&apos;t support dark mode!</>;
  }
}

export default DarkMode;

/*

// The browser supports prefers-color-scheme,
// but cannot find a light/dark preference 
@media (prefers-color-scheme: no-preference) {
  background: black;
  color: white;
}

// Use dark mode
@media (prefers-color-scheme: dark) {
  background: black;
  color: white;
}

// Use light mode 
@media (prefers-color-scheme: light) {
  background: white;
  color: black;
}

*/
