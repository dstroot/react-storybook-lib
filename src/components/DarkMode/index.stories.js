import React from 'react';
import { storiesOf } from '@storybook/react';
import DarkMode from '../DarkMode';

storiesOf('DarkMode', module).add('show DarkMode', () => <DarkMode />);
