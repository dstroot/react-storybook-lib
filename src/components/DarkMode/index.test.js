import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import DarkMode from '../DarkMode';

describe('DarkMode', () => {
  // mount the component
  let mountedComponent;
  const getMountedComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<DarkMode />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('it should render', () => {
    let tree = create(<DarkMode />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
