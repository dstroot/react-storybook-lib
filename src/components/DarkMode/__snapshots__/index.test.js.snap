// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkMode it should render 1`] = `"Your browser doesn't support dark mode!"`;
